code-challenges.test.js

```diff
@@ -13,47 +16,151 @@
 
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
-
+//input= array of words
+//Expected output= array with all the words capitalized
+// i plan on making a const that will take in an array of words and using a for loop return the array with the words in caps.
+// const capArray = array.map((element) => {
+//   return element.toUpperCase();
+// });
 // a) Create a test with expect statements for each of the variables provided.
-
-const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+//created the parameters of the test to include what are we testing and the expected output to be. as i stated earlier, I'm not sure why jest is not functioning properly. but i hope the rest of my answer is correct.
+describe("capArray", () => {
+  it("takes an array of words and returns an array with the same words capitalized", () => {
+    expect(capArray(randomNouns1)).toEqual(
+      "Streetlamp",
+      "Potato",
+      "Teeth",
+      "Conclusion",
+      "Nephew"
+    );
+    expect(capArray(randomNouns2)).toEqual(
+      "Temperature",
+      "Database",
+      "Chopsticks",
+      "Mango"
+    );
+  });
+});
+
+const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"];
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
-const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
-// Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+// i will plug this array into my function
 
+//console.log(capArray(randomNouns1));
+// this returns a typeerror which i am lead to believe the current command is being used in the wrong situation or in the wrong spot of the code. will move forward.
+// after the changes i got the typeerror that states .toUpperCase() is not a function. not sure how to fix that and the syntax seems fine. but obviously im wrong here.
+//now after my changes it returns that array.map.toUpperCase() isn't a function. not sure what i might be over looking or over thinking it.
+const randomNouns2 = ["temperature", "database", "chopsticks", "mango"];
+// Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+//i will write the test for this even though my jest doesnt seem to be aggreeing with me.
+describe("capArray", () => {
+  it("takes an array with strings and returns an array with the strings capitalized", () => {
+    expect(capArray()).toEqual(array.toUpperCase());
+  });
+});
 // b) Create the function that makes the test pass.
-
+// reusing the function i created. i believe i have hit the point in my testing that is running into syntax and operating problems due to the information in this file. i will restart terminal and comment out all information that doesn't obtain to this specific test..
+// const capArray = (array) => {
+//   return array().map("string").toUpperCase();
+// };
```

Comment:
Map is a great start to this problem.  Keep in mind you will need to also break up the first character in each string and capitalize only that, then join it back to the rest of it's word.